### PR TITLE
Fix some error by update table view and control area function

### PR DESCRIPTION
**Pull request by TruongQuangSB**
- Fix:
  - Table not rendere:  when Table X open -> Table Y open -> Tabelle X close -> Table Y close -> Table X open
  - ControlArea value reset to default, when change table type
  - Update table two times, when table type change
  - Wrong compare content

### CODE REVIEW

File: java/bundles/org.eclipse.set.application/src/org/eclipse/set/application/controlarea/ControlAreaSelectionControl.java [hunk 1]
```diff
@@ -118,8 +122,10 @@ private void createCombo(final Composite parent) {
 			@Override
 			public void accept(final NewTableTypeEvent t) {
 				initCombo();
-				tableType = t.getTableType();
-				setCombo();
+				setCombo(t.getTableType());
+				// Send update event, when table type change
+				seletcionControlArea(comboViewer.getSelection(),
+						t.getTableType());
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.application/src/org/eclipse/set/application/controlarea/ControlAreaSelectionControl.java:128]: trigger update table event, when table type change

File: java/bundles/org.eclipse.set.application/src/org/eclipse/set/application/controlarea/ControlAreaSelectionControl.java [hunk 2]
```diff
@@ -171,16 +177,34 @@ private void setCombo() {
...
+						return areaValue.areaName()
+								.equals(oldAreaValue.areaName());
+					}
+					return false;
+				}).findFirst();
+		if (!oldSelectionValue.equals(messages.ControlAreaCombo_Default_Value)
+				&& oldValue.isPresent()) {
+			final int index = comboViewer.getCombo()
+					.indexOf(oldValue.get().areaName());
+			comboViewer.getCombo().select(index);
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.application/src/org/eclipse/set/application/controlarea/ControlAreaSelectionControl.java:204]: Set selection to old value, when table type change

File: java/bundles/org.eclipse.set.feature.table/src/org/eclipse/set/feature/table/ToolboxTableView.java [hunk 3]
```diff
@@ -186,13 +185,6 @@ public TableType getTableType() {
 		return tableType;
 	}
 
-	@Override
-	public void handleNewTableType(final NewTableTypeEvent e) {
-		tableType = e.getTableType();
-		controlAreas.clear();
-		updateTableView();
-	}
-
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table/src/org/eclipse/set/feature/table/ToolboxTableView.java:195]: when table type change, the event of controlarea will trigger table update

File: java/bundles/org.eclipse.set.feature.table/src/org/eclipse/set/feature/table/internal/TableServiceImpl.java [hunk 4]
```diff
@@ -263,19 +269,24 @@ private void saveTableError(final String shortCut,
...
-					modelSession.getContainer(tableType.getContainerForTable()),
-					controlArea);
+			final MultiContainer_AttributeGroup container = modelSession
+					.getContainer(tableType.getContainerForTable());
+			final Stell_Bereich area = StreamSupport
+					.stream(container.getStellBereich().spliterator(), false)
+					.filter(e -> e.getIdentitaet().getWert()
+							.equals(controlAreaId))
+					.findFirst().orElse(null);
+			transformedTable = modelService.transform(container, area);
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table/src/org/eclipse/set/feature/table/internal/TableServiceImpl.java:289]: Find reference Stell_Bereich in target container

**Review by pstuecker — APPROVED** [review]

**Pull request merged**